Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -737,6 +739,24 @@ public static class Extensions {
             return (EnvDTE.DTE)provider.GetService(typeof(EnvDTE.DTE));
         }
 
+        internal static SVsShellDebugger GetShellDebugger(this IServiceProvider provider) {
+            return (SVsShellDebugger)provider.GetService(typeof(SVsShellDebugger));
+        }
+
+        internal static async System.Threading.Tasks.Task RefreshVariableViews(this IServiceProvider serviceProvider) {
+            EnvDTE.Debugger debugger = serviceProvider.GetDTE().Debugger;
+            foreach (AD7Engine engine in AD7Engine.GetEngines()) {
+                if (engine.Process != null && engine.Process.Id == debugger.CurrentProcess.ProcessID) {
+                    await engine.RefreshThreadFrames(debugger.CurrentThread.ID);
```

Comment:
What happens in the parallel watch window? I suspect that we can only modify values on the current thread anyway, but worth checking in case we need to refresh all threads.